tests/integration/bond_test.yaml

```diff
@@ -0,0 +1,375 @@
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-tlb
+    options:
+      tlb_dynamic_lb: true
+description: Delete dummy0
+interfaces:
+- name: dummy0
+  type: dummy
+  state: absent
+description: Bring up dummy0 device
+interfaces:
+- name: dummy0
+  type: dummy
+  state: up
+description: Set the bonding mode to balance-tlb for bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-tlb
+description: Disable tlb_dynamic_lb for bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    options:
+      tlb_dynamic_lb: false
+description: Set the bonding mode balance-rr and miimon value to 140
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-rr
+    options:
+      miimon: 140
+description: Set the bonding mode to balance-tlb for bond device bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-tlb
+description: Create the vlan with vlan ID 200 over bond99
+interfaces:
+- name: bond99.200
+  type: vlan
+  state: up
+  vlan:
+    base-iface: bond99
+    id: 200
+description: Set the bonding mode to active-backup for bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: active-backup
+description: Remove the vlan bond99.200
+interfaces:
+- name: bond99.200
+  type: vlan
+  state: absent
+description: Set arp_missed_max to 200 for bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    options:
+      arp_missed_max: 200
+description: Set miimon of the bonding device bond99 to 200
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    options:
+      miimon: 200
+description: Set the xmit_hash_policy to layer2+3 and bonding mode to balance-xor for bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-xor
+    options:
+      xmit_hash_policy: layer2+3
+description: Set the mtu of eth1 to 1280
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  mtu: 1280
+description: Bring up eth1 device
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+description: Remove bond device bond99, bring down port eth1 and eth2
+interfaces:
+- name: bond99
+  type: bond
+  state: absent
+- name: eth1
+  type: ethernet
+  state: down
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+  controller: ''
+- name: eth2
+  type: ethernet
+  state: down
+  controller: ''
+description: Remove all ports from the bond device bond99
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    port: []
+description: Remove bond99, configure the empty controller, the address 192.168.1.1/24, 2001:db8:1::1/64, dhcp4 disabled, dhcp6 disabled, autoconf disabled for eth1 ethernet device
```

Comment:
how about `detach eth1 from controller` instead of `configure the empty controller`.